### PR TITLE
ISPN-13233 configuration doc updates

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-13233

The diff is probably a little scary: +9,376 lines. But this PR overhauls the Configuration Guide with the following changes:

- Adds clustered cache mode diagrams from an illustrator for both community and downstream.
- Adds diagram for off-heap memory (JVM memory space) and removes the bucket array with address link pointers. Dan and Wolf both agreed in review it was too low level. 
- Re-organizes cache mode assemblies so topics map to the correct information type: concept, task, reference.
- Removes broken links and extraneous information. Some of the content for cache modes 
- s/replication mode/replicated caches and s/distributed mode/distributed caches
- Includes an editing pass for local and clustered cache mode topics.
- Changes several headings to improve readability and scan-ability. (More than three words in titles, include keywords that match configuration parameters...)
- Changes headings to sentence casing.
- Updates to procedures for configuring eviction, expiration, and off-heap memory.
- Added configuration examples under discrete second level headings to include XML,JSON,YAML, and ConfigurationBuilder.

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/code_examples/EvictionPassivation.java [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ConfigurationBuilder builder = new ConfigurationBuilder();
+builder.memory().maxCount(100);
+builder.persistence().passivation(true);//Persistent storage configuration
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/topics/code_examples/EvictionPassivation.java:3]: the lack of space before the comment irks me :)

File: documentation/src/main/asciidoc/topics/con_capacity_factors.adoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+To configure capacity factors, you specify any non-negative number and the
+{brandname} hashing algorithm assigns each node a load weighted by its
+capacity factor (both as a primary owner and as a backup owner).
+
+For example, nodeA has 2x the memory available than nodeB in the same
+{brandname} cluster. In this case, setting `capacityFactor` to a value of `2`
+configures {brandname} to allocate 2x the number of segments to nodeA.
+
+Setting a capacity factor of `0` is possible but is recommended only in cases
+where nodes are not joined to the cluster long enough to be useful data owners.
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/topics/con_capacity_factors.adoc:15]: Maybe mention that cf can be fractional

**Comment by oraNod** [comment]
> Did you really mean to add `documentation/src/main/asciidoc/titles/configuring/configuring.html ` ?

No, I'm not entirely sure how that got in there. I have gitignore configured for .html files. Will remove, thanks for the catch.

File: documentation/src/main/asciidoc/topics/code_examples/EvictionPassivation.java (hunk 1 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/code_examples/EvictionPassivation.java:3]: fixed

**Review by karesti — APPROVED** [review]

**Comment by karesti** [comment]
@oraNod great work!

**Comment by tristantarrant** [comment]
Merged

**Pull request merged**